utils/tx/tx.go

```diff
@@ -6,103 +6,182 @@ import (
 	"encoding/json"
 	"fmt"
 	"net/http"
+	"strings"
 	"time"
 
 	cometclient "github.com/cometbft/cometbft/rpc/client/http"
 	comettypes "github.com/cometbft/cometbft/types"
+	"github.com/gorilla/websocket"
 	"github.com/pterm/pterm"
 )
 
 func MonitorTransaction(wsURL, txHash string) error {
-	for {
-		_, err := http.Get(fmt.Sprintf("%s/status", wsURL))
-		if err == nil {
-			fmt.Println("✅ RPC is working!")
-			break
+	if strings.HasPrefix(wsURL, "http") {
+		for {
+			_, err := http.Get(fmt.Sprintf("%s/status", wsURL))
+			if err == nil {
+				fmt.Println("✅ RPC is working!")
```

Comment:
can you replace these checks with pterm's loader to have the same visual output across roller, eg

```
	spinner, _ := pterm.DefaultSpinner.Start("checking rpc availability")
```

also, this seems generic anough to be a util function, like `WaitForHealthyRollApp` that we already have

Reply:
replaced